batch-jberet/pom.xml

```diff
@@ -75,6 +75,10 @@
             <groupId>org.jboss.msc</groupId>
             <artifactId>jboss-msc</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.picketbox</groupId>
+            <artifactId>picketbox</artifactId>
```

Comment:
Should this have `<optional>true</optional>`?

Reply:
I suspect not as it will not compile without it - but maybe I should set the scope to provided to stop it leaking out.

Reply:
I think optional is still valid as it's optional at runtime. If the legacy security is desired, then the dependency is not optional. However it should work at runtime without it.

Reply:
TBH here we are talking about class loading for a subsystem, by that point the pom definition has been dropped and we have moved to module definitions for class loading.  I will add scope provided to all the picketbox dependencies as I add them so we can avoid transitive dependencies creeping in.  I will add optional if you really want it but as we don't use maven dependencies at runtime I think that is redundant information.

Reply:
I've got no real strong opinion either way really. We do use `<optional>true</optional>` for the log tooling dependencies. I don't recall why we decided to do that though.